packages/factory_reset_tools/lib/dbus/drive_manager.dart

```diff
@@ -0,0 +1,203 @@
+          signature: DBusSignature('ao'),
+        );
+        for (final partition in partitions.asObjectPathArray()) {
+          blockDevs.remove(partition);
+        }
+      } on DBusInvalidArgsException {
+        // partition table might not be in the block device
+        continue;
+      }
+    }
+
+    // finally we got a list of drives, find those who's ConnectionBus is "usb",
+    // or is Removable, since USB NVMe drives does not show Removable
+    final drives = <DriveData>[];
+    for (final entry in blockDevs.entries) {
+      final blockDeviceObjectPath = entry.key;
+      final driveObjectPath = entry.value;
+
+      final driveObject = DBusRemoteObject(
+        _dbusClient,
+        name: 'org.freedesktop.UDisks2',
+        path: driveObjectPath,
+      );
+      final connectionBus = await driveObject.getProperty(
+        'org.freedesktop.UDisks2.Drive',
+        'ConnectionBus',
+        signature: DBusSignature('s'),
+      );
+      final isRemovable = await driveObject.getProperty(
+        'org.freedesktop.UDisks2.Drive',
+        'Removable',
+        signature: DBusSignature('b'),
+      );
+
+      if (!(connectionBus.asString() == 'usb' || isRemovable.asBoolean())) {
+        continue;
+      }
+
+      final blockDeviceObject = DBusRemoteObject(
+        _dbusClient,
+        name: 'org.freedesktop.UDisks2',
+        path: blockDeviceObjectPath,
+      );
+      final id = await driveObject.getProperty(
+        'org.freedesktop.UDisks2.Drive',
+        'Id',
+        signature: DBusSignature('s'),
+      );
+      final vendor = await driveObject.getProperty(
+        'org.freedesktop.UDisks2.Drive',
+        'Vendor',
+        signature: DBusSignature('s'),
+      );
+      final model = await driveObject.getProperty(
+        'org.freedesktop.UDisks2.Drive',
+        'Model',
+        signature: DBusSignature('s'),
+      );
+      final size = await driveObject.getProperty(
+        'org.freedesktop.UDisks2.Drive',
+        'Size',
+        signature: DBusSignature('t'),
+      );
+      final devicePath = await blockDeviceObject.getProperty(
+        'org.freedesktop.UDisks2.Block',
+        'Device',
+        signature: DBusSignature('ay'),
+      );
+      // Note that Device property is an array of unsigned bytes, containing
+      // null-terminated C-style string, therefore needs to filter out 0 from
+      // the byte array.
+      final devicePathString = String.fromCharCodes(
+        devicePath.asByteArray().where((e) => e != 0),
+        0,
+        128,
+      );
+      drives.add(
+        DriveData(
+          '${vendor.asString()} ${model.asString()}'.trim(),
+          id.asString(),
+          size.asUint64(),
+          devicePathString,
+          driveObjectPath,
+          blockDeviceObjectPath,
+        ),
+      );
+    }
+    drives.sort((a, b) => a.id.compareTo(b.id));
+    return drives;
+  }
+
+  static Stream<ResetMediaCreationProgress> dummyCreateResetMedia(
```

Comment:
Nitpick: `s/dummy/fake`

Reply:
Thanks, it was my placeholder code for testing the UI